lib/Runtime/Library/DataView.cpp

```diff
@@ -517,14 +517,14 @@ namespace Js
 
         if (args.Info.Count == 0 || !VarIs<DataView>(args[0]))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedDataView,  _u("offset or value"));
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedDataView);
         }
-        if (args.Info.Count < 3)
+        if (args.Info.Count < 2)
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_DataView_NeedArgument);
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_DataView_NeedArgument, _u("offset"));
         }
         DataView* dataView = VarTo<DataView>(args[0]);
-        float value = JavascriptConversion::ToFloat(args[2], scriptContext);
+        float value = args.Info.Count == 2 ? static_cast<float>(JavascriptNumber::NaN) : JavascriptConversion::ToFloat(args[2], scriptContext);
```

Comment:
Nit: let's start to tame the overlong line lengths in the codebase for readability:

```cpp
float value = args.Info.Count == 2
    ? static_cast<float>(JavascriptNumber::NaN)
    : JavascriptConversion::ToFloat(args[2], scriptContext);
```

Reply:
Done.